source/CardGames.py

```diff
@@ -184,36 +173,28 @@ def High(self):
   def clearHand(self):
     self.hand = []
     self.knownCards = []
-  
-  ##MY FUNCTION##
-
-  def printHand(self):
-    list1 = []
-    for i in self.hand:
-      list1.append(str(i))
-    return list(map(int, list1))
-
-  def get_playerHand(self):
-    print()
-
-  def addValues(self):
-    list1 = []
-    #Creates empty list
-    for i in self.hand:
-      #print(i)
-      #gets each value of card that is found in self.hand
-      list1.append(str(i))
-      #appends the value as string to a list
-    integer_list = list(map(int, list1))
-    #converts all strings in list into integer
-    return sum(integer_list)
+  def printCardsIfSet(self):
+    # Check for a set of four cards with the same value
+    value_counts = {}
+    for card in self.hand:
+      if card.value not in value_counts:
+        value_counts[card.value] = 1
+      else:
+        value_counts[card.value] += 1
+      if value_counts[card.value] == 4:
+        print("You have a set of four %s's!" % card.value)
+        print("Would you like to print your hand? (Y/N)")
+        choice = input().lower()
+        if choice == 'y':
+          self.showHand()
+        return
 
```

Comment:
I like that you created a print statement to display if someone has a set of four! Well done!

Reply:
Thanks, Peter!